modules/nw-sriov-cfg-bond-interface-with-virtual-functions.adoc

```diff
@@ -0,0 +1,110 @@
+
+[id="nw-sriov-cfg-bond-interface-with-virtual-functions_{context}"]
+= Configuring a bond interface from two SR-IOV interfaces
+
+Bonding enables multiple network interfaces to be aggregated into a single logical "bonded" interface. Bond-CNI brings bond capability into containers.
+
+Bond-CNI can be created using SR-IOV virtual functions and placing them in the container network namespace.
+
+Red Hat OpenShift Platform only supports Bond-CNI using SR-IOV virtual functions. Physical interfaces are not supported.
+
+== Pre-requisites
+
+The SR-IOV operator must be installed  and configured in order to obtain virtual functions in a container.
+
+To configure SR-IOV interfaces, an SR-IOV network and policy must be created for each interface.
+
+The SR-IOV operator creates a network attachment definition for each SR-IOV interface, based on the SR-IOV network and policy defined.
+
+== Creating a bond network attachment definition
+
+Now that the SR-IOV virtual functions are available, you can create a bond network attachment definition.
+
+[source,yaml]
+----
+    apiVersion: "k8s.cni.cncf.io/v1"
+    kind: NetworkAttachmentDefinition
+    metadata:
+      name: bond-net1
+      namespace: demo
+    spec:
+      config: '{
+      "type": "bond", <1>
+      "cniVersion": "0.3.1",
+      "name": "bond-net1",
+      "ifname: "bond0" <2>
+      "mode": "active-backup",
+      "failOverMac": 1,
+      "linksInContainer": true,
+      "miimon": "100",
+      "links": [ <3>
+            {"name": "net1"},
+            {"name": "net2"}
+        ],
+      "ipam": {
+            "type": "host-local"
+            "subnet": "10.56.217.0/24"
+            "routes": [{
+            "dst": "0.0.0.0/0"
+            }],
+            "gateway": "10.56.217.1"
+        }
+      }'
+
+----
+
+<1> The type is "bond".
+
+<2> The ifname attribute specifies the name of the bond interface.
+
+<3> The links section defines which interfaces will be used to create the bond. Multus names the attached interfaces as: "net", plus a consecutive number, starting with one.
+
+== Creating a pod using a bond interface
+
+You can now test the setup by creating a pod using a bond interface.
+
+[source,yaml]
+----
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: bondpod1
+      namespace: mm
+      annotations: <1>
+        k8s.v1.cni.cncf.io/networks: mm/sriovnet1, mm/sriovnet2, mm/bond-net <2>
+    spec:
+      containers:
+      - name: podexample
+        image: quay.io/openshift/origin-network-interface-bond-cni:4.9.0
+        command: ["/bin/bash", "-c", "sleep INF"]
+
+----
+
+<1> Note the network annotation.
+
+<2> The network annotation contains two SR-IOV network attachments, and one bond network attachment. The bond attachment uses the two SR-IOV interfaces as bond slaves.
+
+You can inspect the pod interfaces with the following command:
+
+[source,yaml]
+----
+$ oc rsh -n mm bondpod1
+sh-4.4#
+sh-4.4# ip l
```

Comment:
The output is incorrect - there are no slaves, there must be net1 and net2
example:
https://github.com/openshift/bond-cni/blob/master/README.md?plain=1#L247

Reply:
Change command to sh-4.4# ip a 
from 
sh-4.4# ip l  and use this output:

1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
3: eth0@if150: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1450 qdisc noqueue state UP 
    link/ether 62:b1:b5:c8:fb:7a brd ff:ff:ff:ff:ff:ff
    inet 10.244.1.122/24 brd 10.244.1.255 scope global eth0
       valid_lft forever preferred_lft forever
4: bond0: <BROADCAST,MULTICAST,UP,LOWER_UP400> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether 9e:23:69:42:fb:8a brd ff:ff:ff:ff:ff:ff
    inet 10.56.217.66/24 scope global bond0
       valid_lft forever preferred_lft forever
43: net1: <BROADCAST,MULTICAST,UP,LOWER_UP800> mtu 1500 qdisc mq master bond0 state UP qlen 1000
    link/ether 9e:23:69:42:fb:8a brd ff:ff:ff:ff:ff:ff
44: net2: <BROADCAST,MULTICAST,UP,LOWER_UP800> mtu 1500 qdisc mq master bond0 state UP qlen 1000
    link/ether 9e:23:69:42:fb:8a brd ff:ff:ff:ff:ff:ff